Dialog/SpritesEditorDialog.cpp

```diff
@@ -0,0 +1,561 @@
+﻿#include "SpritesEditorDialog.h"
+#include "ui_SpritesEditorDialog.h"
+
+#include <QStringList>
+#include <QScrollBar>
+#include <QMessageBox>
+#include <QInputDialog>
+#include <QColorDialog>
+#include <QFileDialog>
+#include "ROMUtils.h"
+#include "FileIOUtils.h"
+
+/// <summary>
+/// Constructor of SpritesEditorDialog class.
+/// </summary>
+SpritesEditorDialog::SpritesEditorDialog(QWidget *parent, DialogParams::EntitiesAndEntitySetsEditParams *entitiesAndEntitySetsEditParams) :
+    QDialog(parent),
+    ui(new Ui::SpritesEditorDialog),
+    entitiesAndEntitySetsEditParam(entitiesAndEntitySetsEditParams)
+{
+    ui->setupUi(this);
+
+    // Render Setup
+    SpriteTileMAPScene = new QGraphicsScene(0, 0, 8 * 32, 8 * 32);
+    ui->graphicsView_SpriteTileMap->setScene(SpriteTileMAPScene);
+    ui->graphicsView_SpriteTileMap->setAlignment(Qt::AlignTop | Qt::AlignLeft);
+    ui->graphicsView_SpriteTileMap->scale(2, 2);
+    ui->graphicsView_SpriteTileMap->SetCurrentSpritesEditor(this);
+    SpritesetTileMAPScene = new QGraphicsScene(0, 0, 8 * 32, 8 * 32);
+    ui->graphicsView_SpritesetTileMap->setScene(SpritesetTileMAPScene);
+    ui->graphicsView_SpritesetTileMap->setAlignment(Qt::AlignTop | Qt::AlignLeft);
+    PaletteBarScene = new QGraphicsScene(0, 0, 16 * 8, 16 * 8);
+    // TODO: PaletteBarScene->setBackgroundBrush();
```

Comment:
What is this for?

Reply:
perhaps don't need, i just want it to show the extra rows without graphics on it